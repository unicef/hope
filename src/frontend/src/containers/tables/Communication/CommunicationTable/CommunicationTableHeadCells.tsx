import { HeadCell } from '@components/core/Table/EnhancedTableHead';
import { CommunicationMessageNode } from '@generated/graphql';

export const headCells: HeadCell<CommunicationMessageNode>[] = [
  {
    disablePadding: false,
    label: 'Message ID',
    id: 'id',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Title',
    id: 'title',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Number of Recipients',
    id: 'numberOfRecipients',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Created by',
    id: 'created_by',
    numeric: false,
  },
  {
    disablePadding: false,
    label: 'Creation Date',
    id: 'created_at',
    numeric: false,
  },
];
