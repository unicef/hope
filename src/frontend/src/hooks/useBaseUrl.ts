import { useBusinessArea } from './useBusinessArea';
import { useGlobalProgram } from './useGlobalProgram';

export function useBaseUrl(): {
  baseUrl: string;
  programId: string;
  businessArea: string;
  isAllPrograms: boolean;
  isGlobal: boolean;
} {
  const businessArea = useBusinessArea();
  const programId = useGlobalProgram();
  const baseUrl = `${businessArea}/programs/${programId}`;
  const isAllPrograms = programId === 'all';
  const isGlobal = businessArea === 'global';

  return {
    businessArea, programId, baseUrl, isAllPrograms, isGlobal,
  };
}
