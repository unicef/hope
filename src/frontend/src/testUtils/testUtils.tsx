import { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, RenderOptions } from '@testing-library/react';
import { Formik } from 'formik';
import noop from 'lodash/noop';
import { TestProviders } from './testProviders';
import { ProgramStatus } from '@generated/graphql';
import { MockLink } from '@apollo/client/testing';

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) =>
  render(
    <>
      <BrowserRouter>
        <Formik initialValues={{}} onSubmit={noop}>
          {ui}
        </Formik>
      </BrowserRouter>
    </>,
    {
      wrapper: TestProviders,
      ...options,
    },
  );

export * from '@testing-library/react';
export { customRender as render };

let m_w = 123456789;
let m_z = 987654321;
const mask = 0xffffffff;

// Takes any integer
export const seed = (i) => {
  m_w = (123456789 + i) & mask;
  m_z = (987654321 - i) & mask;
};

// Returns number between 0 (inclusive) and 1.0 (exclusive),
// just like Math.random().
export const random = () => {
  m_z = (36969 * (m_z & 65535) + (m_z >> 16)) & mask;
  m_w = (18000 * (m_w & 65535) + (m_w >> 16)) & mask;
  let result = ((m_z << 16) + (m_w & 65535)) >>> 0;
  result /= 4294967296;
  return result;
};

export class ApolloLoadingLink extends MockLink {
  constructor() {
    super([]);
  }

  request() {
    return null;
  }
}

export const fakeContextProgram = {
  selectedProgram: {
    id: '1',
    name: 'someName',
    status: ProgramStatus.Active,
    pduFields: null,
    dataCollectingType: {
      id: '1',
      householdFiltersAvailable: true,
      individualFiltersAvailable: true,
      label: 'data collecting type',
      code: '123',
      type: 'full',
      children: null,
    },
  },
  setSelectedProgram: () => {},
  isActiveProgram: true,
  isSocialDctType: false,
  isStandardDctType: true,
  programHasPdu: false,
};
