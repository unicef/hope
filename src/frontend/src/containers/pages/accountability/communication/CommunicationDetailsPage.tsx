import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { useParams } from 'react-router-dom';
import { Box } from '@mui/material';
import { BreadCrumbsItem } from '@components/core/BreadCrumbs';
import { LoadingComponent } from '@components/core/LoadingComponent';
import { PageHeader } from '@components/core/PageHeader';
import { PermissionDenied } from '@components/core/PermissionDenied';
import { hasPermissions, PERMISSIONS } from '../../../../config/permissions';
import { usePermissions } from '@hooks/usePermissions';
import { isPermissionDeniedError } from '@utils/utils';
import { useAccountabilityCommunicationMessageQuery } from '@generated/graphql';
import { UniversalActivityLogTable } from '../../../tables/UniversalActivityLogTable';
import { CommunicationDetails } from '@components/accountability/Communication/CommunicationDetails';
import { CommunicationMessageDetails } from '@components/accountability/Communication/CommunicationMessageDetails';
import { RecipientsTable } from '../../../tables/Communication/RecipientsTable/RecipientsTable';
import { useBaseUrl } from '@hooks/useBaseUrl';
import { AdminButton } from '@core/AdminButton';

export function CommunicationDetailsPage(): React.ReactElement {
  const { t } = useTranslation();
  const { id } = useParams();
  const { baseUrl } = useBaseUrl();
  const { data, loading, error } = useAccountabilityCommunicationMessageQuery({
    variables: { id },
    fetchPolicy: 'cache-and-network',
  });
  const permissions = usePermissions();

  if (loading) return <LoadingComponent />;

  if (isPermissionDeniedError(error)) return <PermissionDenied />;

  if (!data || permissions === null) return null;

  const message = data.accountabilityCommunicationMessage;

  const breadCrumbsItems: BreadCrumbsItem[] = [
    {
      title: t('Communication'),
      to: `/${baseUrl}/accountability/communication`,
    },
  ];
  return (
    <div>
      <PageHeader
        title={`${message.unicefId}`}
        breadCrumbs={
          hasPermissions(
            PERMISSIONS.ACCOUNTABILITY_COMMUNICATION_MESSAGE_VIEW_DETAILS,
            permissions,
          )
            ? breadCrumbsItems
            : null
        }
        flags={<AdminButton adminUrl={message.adminUrl} />}
      />
      <Box display="flex" flexDirection="column">
        <CommunicationDetails message={message} />
        <CommunicationMessageDetails message={message} />
        <RecipientsTable
          canViewDetails={hasPermissions(
            PERMISSIONS.ACCOUNTABILITY_COMMUNICATION_MESSAGE_VIEW_DETAILS,
            permissions,
          )}
          id={id}
        />
        {hasPermissions(
          PERMISSIONS.ACCOUNTABILITY_COMMUNICATION_MESSAGE_VIEW_DETAILS,
          permissions,
        ) && <UniversalActivityLogTable objectId={id} />}
      </Box>
    </div>
  );
}
