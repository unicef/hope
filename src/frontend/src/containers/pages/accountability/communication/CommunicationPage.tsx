import * as React from 'react';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Link, useLocation } from 'react-router-dom';
import { useGrievancesChoiceDataQuery } from '@generated/graphql';
import { CommunicationFilters } from '@components/accountability/Communication/CommunicationTable/CommunicationFilters';
import { LoadingComponent } from '@components/core/LoadingComponent';
import { PageHeader } from '@components/core/PageHeader';
import { PermissionDenied } from '@components/core/PermissionDenied';
import {
  PERMISSIONS,
  hasPermissionInModule,
} from '../../../../config/permissions';
import { usePermissions } from '@hooks/usePermissions';
import { getFilterFromQueryParams } from '@utils/utils';
import { CommunicationTable } from '../../../tables/Communication/CommunicationTable';
import { useBaseUrl } from '@hooks/useBaseUrl';
import { ButtonTooltip } from '@components/core/ButtonTooltip';
import { useProgramContext } from '../../../../programContext';

export function CommunicationPage(): React.ReactElement {
  const { baseUrl } = useBaseUrl();
  const permissions = usePermissions();
  const location = useLocation();
  const { t } = useTranslation();
  const { isActiveProgram } = useProgramContext();

  const initialFilter = {
    createdBy: '',
    createdAtRangeMin: null,
    createdAtRangeMax: null,
    targetPopulation: '',
  };

  const [filter, setFilter] = useState(
    getFilterFromQueryParams(location, initialFilter),
  );
  const [appliedFilter, setAppliedFilter] = useState(
    getFilterFromQueryParams(location, initialFilter),
  );
  const { data: choicesData, loading: choicesLoading } =
    useGrievancesChoiceDataQuery();

  if (choicesLoading) return <LoadingComponent />;
  if (permissions === null) return null;
  if (
    !hasPermissionInModule(
      PERMISSIONS.ACCOUNTABILITY_COMMUNICATION_MESSAGE_VIEW_LIST,
      permissions,
    )
  )
    return <PermissionDenied />;
  if (!choicesData) return null;

  return (
    <>
      <PageHeader title={t('Communication')}>
        <ButtonTooltip
          variant="contained"
          color="primary"
          component={Link}
          to={`/${baseUrl}/accountability/communication/create`}
          data-cy="button-communication-create-new"
          title={t('Programme has to be active to create new Message')}
          disabled={!isActiveProgram}
        >
          {t('New message')}
        </ButtonTooltip>
      </PageHeader>
      <CommunicationFilters
        filter={filter}
        setFilter={setFilter}
        initialFilter={initialFilter}
        appliedFilter={appliedFilter}
        setAppliedFilter={setAppliedFilter}
      />
      <CommunicationTable
        filter={appliedFilter}
        canViewDetails={hasPermissionInModule(
          PERMISSIONS.ACCOUNTABILITY_COMMUNICATION_MESSAGE_VIEW_LIST,
          permissions,
        )}
      />
    </>
  );
}
