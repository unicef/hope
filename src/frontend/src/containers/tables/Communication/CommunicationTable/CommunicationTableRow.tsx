import * as React from 'react';
import TableCell from '@mui/material/TableCell';
import { useNavigate } from 'react-router-dom';
import { ClickableTableRow } from '@components/core/Table/ClickableTableRow';
import { UniversalMoment } from '@components/core/UniversalMoment';
import { renderUserName } from '@utils/utils';
import { CommunicationMessageNode } from '@generated/graphql';
import { BlackLink } from '@components/core/BlackLink';
import { useBaseUrl } from '@hooks/useBaseUrl';

interface CommunicationTableRowProps {
  message: CommunicationMessageNode;
  canViewDetails: boolean;
}

export function CommunicationTableRow({
  message,
  canViewDetails,
}: CommunicationTableRowProps): React.ReactElement {
  const navigate = useNavigate();
  const { baseUrl } = useBaseUrl();
  const messageDetailsPath = `/${baseUrl}/accountability/communication/${message.id}`;
  const handleClick = (): void => {
    navigate(messageDetailsPath);
  };
  return (
    <ClickableTableRow
      hover
      onClick={canViewDetails ? handleClick : undefined}
      role="checkbox"
      key={message.unicefId}
    >
      <TableCell align="left">
        {canViewDetails ? (
          <BlackLink to={messageDetailsPath}>{message.unicefId}</BlackLink>
        ) : (
          message.unicefId
        )}
      </TableCell>
      <TableCell align="left">{message.title}</TableCell>
      <TableCell align="left">{message.numberOfRecipients}</TableCell>
      <TableCell align="left">{renderUserName(message.createdBy)}</TableCell>
      <TableCell align="left">
        <UniversalMoment>{message.createdAt}</UniversalMoment>
      </TableCell>
    </ClickableTableRow>
  );
}
