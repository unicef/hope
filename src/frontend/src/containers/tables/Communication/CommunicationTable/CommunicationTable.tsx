import { ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import {
  AllAccountabilityCommunicationMessagesQueryVariables,
  CommunicationMessageNode,
  useAllAccountabilityCommunicationMessagesQuery,
} from '@generated/graphql';
import { TableWrapper } from '@components/core/TableWrapper';
import { useBaseUrl } from '@hooks/useBaseUrl';
import { dateToIsoString } from '@utils/utils';
import { UniversalTable } from '../../UniversalTable';
import { headCells } from './CommunicationTableHeadCells';
import { CommunicationTableRow } from './CommunicationTableRow';

interface CommunicationTableProps {
  filter;
  canViewDetails: boolean;
}

export function CommunicationTable({
  filter,
  canViewDetails,
}: CommunicationTableProps): ReactElement {
  const { programId } = useBaseUrl();
  const { t } = useTranslation();
  const initialVariables: AllAccountabilityCommunicationMessagesQueryVariables =
    {
      createdAtRange: JSON.stringify({
        min: dateToIsoString(filter.createdAtRangeMin, 'startOfDay'),
        max: dateToIsoString(filter.createdAtRangeMax, 'endOfDay'),
      }),
      program: programId,
      targetPopulation: filter.targetPopulation,
      createdBy: filter.createdBy || '',
    };
  return (
    <TableWrapper>
      <UniversalTable<
      CommunicationMessageNode,
      AllAccountabilityCommunicationMessagesQueryVariables
      >
        title={t('Messages List')}
        headCells={headCells}
        rowsPerPageOptions={[10, 15, 20]}
        query={useAllAccountabilityCommunicationMessagesQuery}
        queriedObjectName="allAccountabilityCommunicationMessages"
        defaultOrderBy="createdAt"
        defaultOrderDirection="desc"
        initialVariables={initialVariables}
        renderRow={(row) => (
          <CommunicationTableRow
            key={row.id}
            message={row}
            canViewDetails={canViewDetails}
          />
        )}
      />
    </TableWrapper>
  );
}
