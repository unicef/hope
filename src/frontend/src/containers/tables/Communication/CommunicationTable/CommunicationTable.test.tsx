import { MockedProvider } from '@apollo/react-testing';
import { act } from 'react';
import * as React from 'react';
import wait from 'waait';
import { fakeApolloAllCommunicationMessages } from '../../../../../fixtures/communication/fakeApolloAllCommunicationMessages';
import { ApolloLoadingLink, render } from '../../../../testUtils/testUtils';
import { CommunicationTable } from './CommunicationTable';
import { useBaseUrl } from '@hooks/useBaseUrl';

describe('containers/tables//Communication/CommunicationTable', () => {
  it('should render with data', async () => {
    const { programId } = useBaseUrl();
    const { container } = render(
      <MockedProvider
        addTypename={false}
        mocks={fakeApolloAllCommunicationMessages}
      >
        <CommunicationTable
          filter={{
            targetPopulation: '',
            createdBy: '',
            first: 10,
            orderBy: '-created_at',
            program: programId,
          }}
          canViewDetails={false}
        />
      </MockedProvider>,
    );
    await act(() => wait(0)); // wait for response

    expect(container).toMatchSnapshot();
  });

  it('should render loading', () => {
    const { programId } = useBaseUrl();
    const { container } = render(
      <MockedProvider
        addTypename={false}
        mocks={fakeApolloAllCommunicationMessages}
      >
        <CommunicationTable
          filter={{
            targetPopulation: '',
            createdBy: '',
            first: 10,
            orderBy: '-created_at',
            program: programId,
          }}
          canViewDetails={false}
        />
      </MockedProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
