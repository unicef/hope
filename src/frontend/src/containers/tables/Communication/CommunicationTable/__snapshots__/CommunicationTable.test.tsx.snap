// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/tables//Communication/CommunicationTable should render loading 1`] = `
<div>
  <div
    class="sc-KXCwU edfDZM"
  >
    <div
      class="sc-hWWBcw glzqeh"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 sc-jHbxoU cxLkyn css-1gtchvp-MuiPaper-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiTableContainer-root sc-jMLmsk css-70zvr5-MuiTableContainer-root"
        >
          <div
            class="sc-cMHOsR BZKmm MuiBox-root css-0"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-16wdp7y-MuiToolbar-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-15aaxeb-MuiTypography-root"
                data-cy="table-title"
              >
                Messages List
              </h6>
            </div>
            <div
              class="sc-cMHOsR BZKmm MuiBox-root css-1ruxp1v"
            />
          </div>
          <table
            class="MuiTable-root sc-cGXZpB sc-kfmGjC jfkrWK hcGOws css-14k3lwa-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1o8lslm-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1e9a4b0-MuiTableCell-root"
                  scope="col"
                >
                  <span
                    class="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-directionAsc sc-gTrWKq fFZOTJ css-7x9vt0-MuiButtonBase-root-MuiTableSortLabel-root"
                    data-cy="table-label"
                    role="button"
                    tabindex="0"
                  >
                    Message ID
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-7ojauk-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDownwardIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1e9a4b0-MuiTableCell-root"
                  scope="col"
                >
                  <span
                    class="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-directionAsc sc-gTrWKq fFZOTJ css-7x9vt0-MuiButtonBase-root-MuiTableSortLabel-root"
                    data-cy="table-label"
                    role="button"
                    tabindex="0"
                  >
                    Title
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-7ojauk-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDownwardIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1e9a4b0-MuiTableCell-root"
                  scope="col"
                >
                  <span
                    class="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-directionAsc sc-gTrWKq fFZOTJ css-7x9vt0-MuiButtonBase-root-MuiTableSortLabel-root"
                    data-cy="table-label"
                    role="button"
                    tabindex="0"
                  >
                    Number of Recipients
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-7ojauk-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDownwardIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1e9a4b0-MuiTableCell-root"
                  scope="col"
                >
                  <span
                    class="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-directionAsc sc-gTrWKq fFZOTJ css-7x9vt0-MuiButtonBase-root-MuiTableSortLabel-root"
                    data-cy="table-label"
                    role="button"
                    tabindex="0"
                  >
                    Created by
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-7ojauk-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDownwardIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1e9a4b0-MuiTableCell-root"
                  scope="col"
                >
                  <span
                    class="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-directionAsc sc-gTrWKq fFZOTJ css-7x9vt0-MuiButtonBase-root-MuiTableSortLabel-root"
                    data-cy="table-label"
                    role="button"
                    tabindex="0"
                  >
                    Creation Date
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-7ojauk-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      data-testid="ArrowDownwardIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                      />
                    </svg>
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root sc-jZCRgm eekLry css-1o8lslm-MuiTableRow-root"
                data-cy="table-row"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium sc-cZztgT kkzxcC css-1dc80h3-MuiTableCell-root"
                  colspan="5"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root sc-jZCRgm eekLry css-1o8lslm-MuiTableRow-root"
                data-cy="table-row"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium sc-cZztgT kkzxcC css-1dc80h3-MuiTableCell-root"
                  colspan="5"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root sc-jZCRgm eekLry css-1o8lslm-MuiTableRow-root"
                data-cy="table-row"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium sc-cZztgT kkzxcC css-1dc80h3-MuiTableCell-root"
                  colspan="5"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root sc-jZCRgm eekLry css-1o8lslm-MuiTableRow-root"
                data-cy="table-row"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium sc-cZztgT kkzxcC css-1dc80h3-MuiTableCell-root"
                  colspan="5"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root sc-jZCRgm eekLry css-1o8lslm-MuiTableRow-root"
                data-cy="table-row"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium sc-cZztgT kkzxcC css-1dc80h3-MuiTableCell-root"
                  colspan="5"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root sc-jZCRgm eekLry css-1o8lslm-MuiTableRow-root"
                data-cy="table-row"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium sc-cZztgT kkzxcC css-1dc80h3-MuiTableCell-root"
                  colspan="5"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root sc-jZCRgm eekLry css-1o8lslm-MuiTableRow-root"
                data-cy="table-row"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium sc-cZztgT kkzxcC css-1dc80h3-MuiTableCell-root"
                  colspan="5"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root sc-jZCRgm eekLry css-1o8lslm-MuiTableRow-root"
                data-cy="table-row"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium sc-cZztgT kkzxcC css-1dc80h3-MuiTableCell-root"
                  colspan="5"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root sc-jZCRgm eekLry css-1o8lslm-MuiTableRow-root"
                data-cy="table-row"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium sc-cZztgT kkzxcC css-1dc80h3-MuiTableCell-root"
                  colspan="5"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root sc-jZCRgm eekLry css-1o8lslm-MuiTableRow-root"
                data-cy="table-row"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium sc-cZztgT kkzxcC css-1dc80h3-MuiTableCell-root"
                  colspan="5"
                >
                  <span
                    class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3sr0vz-MuiSkeleton-root"
                    style="width: 100%; height: 70px;"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="MuiTablePagination-root css-7mt0f-MuiTablePagination-root"
          data-cy="table-pagination"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-nphfye-MuiToolbar-root-MuiTablePagination-toolbar"
          >
            <div
              class="MuiTablePagination-spacer css-1wtxofq-MuiTablePagination-spacer"
            />
            <p
              class="MuiTablePagination-selectLabel css-s09cke-MuiTablePagination-selectLabel"
              id=":r6:"
            >
              Rows per page:
            </p>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-select MuiSelect-root MuiTablePagination-input css-oegngy-MuiInputBase-root-MuiTablePagination-select"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby=":r6: :r5:"
                class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-mmygx2-MuiSelect-select-MuiInputBase-input"
                id=":r5:"
                role="combobox"
                tabindex="0"
              >
                10
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                tabindex="-1"
                value="10"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-1gpuzre-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiTablePagination-displayedRows css-11cfq65-MuiTablePagination-displayedRows"
            >
              0–0 of 0
            </p>
            <div
              class="MuiBox-root css-nkn1dz"
            >
              <button
                aria-label="previous page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                id=":r8:"
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="KeyboardArrowLeftIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6z"
                  />
                </svg>
              </button>
              <button
                aria-label="next page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                id=":r9:"
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`containers/tables//Communication/CommunicationTable should render with data 1`] = `
<div>
  <div
    class="sc-KXCwU edfDZM"
  >
    <div>
      Unexpected error
    </div>
  </div>
</div>
`;
